packages/model-viewer/src/features/scene-graph/material.ts

```diff
@@ -145,68 +122,26 @@ export class Material extends ThreeDOMElement implements MaterialInterface {
         correlatedMaterials,
     );
 
-    this[$clearcoatTexture] = new TextureInfo(
-        onUpdate,
-        TextureUsage.Clearcoat,
-        null,
-        correlatedMaterials,
-    );
-
-    this[$clearcoatRoughnessTexture] = new TextureInfo(
-        onUpdate,
-        TextureUsage.ClearcoatRoughness,
-        null,
-        correlatedMaterials,
-    );
-
-    this[$clearcoatNormalTexture] = new TextureInfo(
-        onUpdate,
-        TextureUsage.ClearcoatNormal,
-        null,
-        correlatedMaterials,
-    );
-
-    this[$sheenColorTexture] = new TextureInfo(
-        onUpdate,
-        TextureUsage.SheenColor,
-        null,
-        correlatedMaterials,
-    );
-
-    this[$sheenRoughnessTexture] = new TextureInfo(
-        onUpdate,
-        TextureUsage.SheenRoughness,
-        null,
-        correlatedMaterials,
-    );
-
-    this[$transmissionTexture] = new TextureInfo(
-        onUpdate,
-        TextureUsage.Transmission,
-        null,
-        correlatedMaterials,
-    );
-
-    this[$thicknessTexture] = new TextureInfo(
-        onUpdate,
-        TextureUsage.Thickness,
-        null,
-        correlatedMaterials,
-    );
-
-    this[$specularTexture] = new TextureInfo(
-        onUpdate,
-        TextureUsage.Specular,
-        null,
-        correlatedMaterials,
-    );
-
-    this[$specularColorTexture] = new TextureInfo(
-        onUpdate,
-        TextureUsage.SpecularColor,
-        null,
-        correlatedMaterials,
-    );
+    const textureInfoFromUsage = (usage: TextureUsage) => {
+      this[$pbrTextures].set(
+          usage,
+          new TextureInfo(
+              onUpdate,
+              usage,
+              null,
+              correlatedMaterials,
+              ));
+    };
+
+    textureInfoFromUsage(TextureUsage.Clearcoat);
+    textureInfoFromUsage(TextureUsage.ClearcoatRoughness);
+    textureInfoFromUsage(TextureUsage.ClearcoatNormal);
+    textureInfoFromUsage(TextureUsage.SheenColor);
+    textureInfoFromUsage(TextureUsage.SheenRoughness);
+    textureInfoFromUsage(TextureUsage.Transmission);
+    textureInfoFromUsage(TextureUsage.Thickness);
+    textureInfoFromUsage(TextureUsage.Specular);
+    textureInfoFromUsage(TextureUsage.SpecularColor);
```

Comment:
💯 
nit: maybe `createTextureInfo`? FromUsage is already clear from the input type.